### PR TITLE
8261862: Expand discussion of rationale for BigDecimal equals/compareTo semantics

### PR DESCRIPTION
**Pull request by jddarcy**
I considered @stuart-marks previous suggestion during the code review of JDK-8261123 to include a more explicit discussion of why, say, different representations of 2 should not be regarded as equivalent. After contemplating several alternatives, I didn't find anything simpler than Stuart's 2/3 example so I used that as seen in the diff.

A short digression, BigDecimal supports both fixed-point style and floating-point style rounding. Floating-point rounding primarily replies on the number of precision digits, regards of their scale, while fixed-point style rounding prioritizes the scale. The number of digits of eventual output is a function of number number of digits in the inputs and the number of precision digits in a floating-point style rounding. A floating-point style rounding has a preferred scale, rather than a fixed scale based on the inputs. The fixed-point style divide method used in the example has a scale based on the dividend, allowing a relatively simple expression to show a distinction between 2.0 and 2.00.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8261862](https://bugs.openjdk.java.net/browse/JDK-8261862): Expand discussion of rationale for BigDecimal equals/compareTo semantics

### Reviewers
 * [Stuart Marks](https://openjdk.java.net/census#smarks) (@stuart-marks - **Reviewer**) ⚠️ Review applies to 48b49386bd49fc387d309483f299d8c4fa7eb994
 * [Brian Burkhalter](https://openjdk.java.net/census#bpb) (@bplb - **Reviewer**) ⚠️ Review applies to 48b49386bd49fc387d309483f299d8c4fa7eb994

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/2804/head:pull/2804`
`$ git checkout pull/2804`

### CODE REVIEW

File: src/java.base/share/classes/java/math/BigDecimal.java [hunk 1]
```diff
@@ -3141,7 +3141,17 @@ else if (ys != INFLATED)
      * compareTo}, this method considers two {@code BigDecimal}
      * objects equal only if they are equal in value and
      * scale. Therefore 2.0 is not equal to 2.00 when compared by this
-     * method.
+     * method since the former has [{@code BigInteger},
+     * {@code scale}] components equal to [20, 1] while the latter has
+     * components equals to [200, 2].
```
Comment (bplb) [inline src/java.base/share/classes/java/math/BigDecimal.java:3146]: s/equals/equal/

File: src/java.base/share/classes/java/math/BigDecimal.java [hunk 2]
```diff
@@ -3141,7 +3141,17 @@ else if (ys != INFLATED)
...
+     *
+     * <p>One example that shows how 2.0 and 2.00 are <em>not</em>
+     * substitutable for each other under some arithmetic operations
+     * either are the two expressions:<br>
+     * {@code new BigDecimal("2.0" ).divide(BigDecimal.valueOf(3),
+     * HALF_UP)} which evaluates to 0.7 and <br>
+     * {@code new BigDecimal("2.00").divide(BigDecimal.valueOf(3),
+     * HALF_UP)} which evaluates to 0.<b>6</b>7.
      *
```
Comment (stuart-marks) [inline src/java.base/share/classes/java/math/BigDecimal.java:3155]: Should this be in an `@apiNote`?

I'm not sure adding the boldface 0.**6**7 is helpful; 0.7 is self-evidently unequal to 0.67.

**Review by stuart-marks — APPROVED** [review]

**Review by bplb — APPROVED** [review]

File: src/java.base/share/classes/java/math/BigDecimal.java (hunk 2 above)
Comment (jddarcy) [reply src/java.base/share/classes/java/math/BigDecimal.java:3156]: Changed as suggested in final version.

File: src/java.base/share/classes/java/math/BigDecimal.java (hunk 1 above)
Comment (jddarcy) [reply src/java.base/share/classes/java/math/BigDecimal.java]: Fixed.

**Pull request closed without merging**